packages/@aws-cdk/aws-rds/lib/private/parameter-group-family-mapping.ts

```diff
@@ -12,3 +14,28 @@ export interface ParameterGroupFamilyMapping {
    */
   readonly parameterGroupFamily: string
 }
+
+/**
+ * Get the latest parameter group family for this engine. Latest is determined using semver on the engine major version.
+ * When `engineVersion` is specified, return the parameter group family corresponding to that engine version.
+ * Return undefined if no parameter group family is defined for this engine or for the requested `engineVersion`.
+ */
+export function calculateParameterGroupFamily(
+  parameterGroupFamilies: ParameterGroupFamilyMapping[] | undefined,
```

Comment:
_(For my own edification):_ Is there any logical difference between a required parameter that can be undefined vs an optional parameter, especially in cases where all parameters are the same (e.g., no issues with optional parameters needing to be the last ones)?

Reply:
There is a tiny difference: for a required parameter that can be undefined, you have to always explicitly pass it (even if just the `undefined` value), while an optional parameter can be skipped from the call altogether.